SwiftyStoreKit/InAppReceipt.swift

```diff
@@ -235,5 +241,69 @@ internal class InAppReceipt {
             }
             task.resume()
     }
-    
+
+    /**
+     *  Verify the purchase of a product in a receipt
+     *  - Parameter productId: the product id of the purchase to verify
+     *  - Parameter inReceipt: the receipt to test in
+     *  - Parameter validUntil: the expires date of the subscription must be valid until this date. If nil, no verification
+     */
+    class func verifyPurchase(
+      productId productId: String,
+      inReceipt receipt: ReceiptInfo,
+      validUntil: NSDate? = nil
+    ) -> SwiftyStoreKit.VerifyPurchaseResult {
+      
+          // Get all receipts
+          guard let allReceipts = receipt["receipt"]?["in_app"] as? [ReceiptInfo] else {
+              return .NotPurchased
+          }
+      
+          // Filter receipts with matching product id
+          let receiptsMatchingProductId = allReceipts
+          .filter { (receipt) -> Bool in
+                let product_id = receipt["product_id"] as? String
+                return product_id == productId
+          }
+      
+          // Verify that at least one receipt has the right product id
+          guard receiptsMatchingProductId.count >= 1 else {
+              return .NotPurchased
+          }
+      
+          guard let validUntil = validUntil else {
+              // Do not need to verify the expiration date
+              return .Purchased(expiresDate: nil)
+          }
+      
+          // Return the expires dates sorted desc
+          let expiresDates = receiptsMatchingProductId
+          .map { (receipt) -> NSDate in
+            // TODO: should use 'expires_date' or 'expiration_date' instead? See:
+            // https://developer.apple.com/library/mac/releasenotes/General/ValidateAppStoreReceipt/Chapters/ReceiptFields.html
+              let expires_date = receipt["expires_date_ms"] as? NSString
+              let expires_date_double = (expires_date?.doubleValue ?? 0.0) / 1000
+              return NSDate(timeIntervalSince1970: expires_date_double)
```

Comment:
I have updated the API so that it takes a `PurchaseType` enum as a parameter and this is used to determine whether the expiration date should be considered. This puts the responsibility on the client to decide which is the correct PurchaseType for each IAP, which I believe is a reasonable request. Should the client specify a `AutomaticallyRenewableSubscription` with a date for a purchase that really is `Consumable`, the completion block will likely say that the purchase is expired, however this can be classified as incorrect usage of the API.